### DISCUSSION TITLE
Codemods skipping with no reason given

### DISCUSSION THREAD

**Discussion by Southclaws** [vercel/next.js#64610] [Ideas]
# Bug report

## Describe the bug

Trying to run the name-default-component codemod but it's just skipping all my files.

## To Reproduce

Clone `https://github.com/openmultiplayer/web`
cd into `frontend/src/components/templates`
run `codemod name-default-component`

## Expected behavior

The components to be transformed into export default function X() ...

## Screenshots

```
❯ codemod name-default-component --force
WARNING: Git directory is not clean. Forcibly continuing.
? On which files or directory should the codemods be applied? .
Executing command: jscodeshift --verbose=2 --ignore-pattern=**/node_modules/** --ignore-pattern=**/.next/** --extensions=tsx,ts,jsx,js --parser=tsx --transform C:\Users\Southclaws\AppData\Roaming\npm\node_modules\@next\codemod\transforms\name-default-component.js .
Processing 7 files... 
Spawning 3 workers...
Sending 3 files to free worker...
Sending 3 files to free worker...
Sending 1 files to free worker...
[BABEL] Note: The code generator has deoptimised the styling of C:\Users\Southclaws\AppData\Roaming\npm\node_modules\@next\codemod\node_modules\lodash\lodash.js as it exceeds the max of 500KB.
[BABEL] Note: The code generator has deoptimised the styling of C:\Users\Southclaws\AppData\Roaming\npm\node_modules\@next\codemod\node_modules\lodash\lodash.js as it exceeds the max of 500KB.
[BABEL] Note: The code generator has deoptimised the styling of C:\Users\Southclaws\AppData\Roaming\npm\node_modules\@next\codemod\node_modules\lodash\lodash.js as it exceeds the max of 500KB.
 SKIP index.js
 SKIP lowercase-note.js
 SKIP npc-callbacks-tip.js
 SKIP version-warning.js
 SKIP id\npc-callbacks-tip.js
 SKIP id\lowercase-note.js
 SKIP id\version-warning.js
All done. 
Results: 
0 errors
0 unmodified
7 skipped
0 ok
Time elapsed: 2.431seconds
```

**Comment by timneutkens** [comment]
https://github.com/openmultiplayer/web/commit/a411e1495adff390bf073f11dd0576c6897bc37c#diff-ba4e88e9a806fe3d7a3d8dc7cc336cefcf0bbec0ca54ba284362b56943e9cbcb

Looking at the code before you made the changes it seems that you tried to run the codemod over functions that could not be detected as being React components as they did not return JSX, potentially this can be made clearer while running the codemod.

**Comment by webdif** [comment]
I have the same problem: when I run `npx @next/codemod name-default-component`, all my .js files are skipped, even when the default export return JSX. Is there anything to do, a param to add or something?

**Comment by yoavaviram** [comment]
Same problem here.

**Comment by markbecker-work** [comment]
Same here

**Comment by ShooTeX** [comment]
@webdif could you show us how your component.js looks like?

**Comment by webdif** [comment]
@ShooTeX I have very simple components, like this:

```js
// component.js
export default ({ prop1, prop2 }) => {
  /* ... */
};
```

**Comment by matt-dalton** [comment]
Still struggling with this...any workarounds anyone's discovered?

**Comment by MarioSimou** [comment]
I had a similar issue. Are there any updates?

**Comment by sebaveg** [comment]
any update?

**Comment by jdetlefs-godaddy** [comment]
I would like to fix this but I am not finding good "Getting Started" notes in https://github.com/vercel/next.js/tree/canary/packages/next-codemod, if someone who has contributed could write some notes on how they test a local problem codebase against a local build of next-codemod that would be super helpful.

**Comment by jdetlefs-godaddy** [comment]
Okay, @timneutkens I'm 95% sure the problem here isn't an issue with the codemod but an issue with the babel plugin that is telling us to run the codemod. That babel plugin is throwing out myriad warnings for functions that dont return JSX and dont qualify for the codemod. 

A fix is in progress here: https://github.com/vercel/next.js/pull/32636/files but I'm struggling with how to test it.

**Comment by jdetlefs-godaddy** [comment]
@Southclaws can you provide an example of a file that returns JSX and should be named but isnt? That would be super helpful.

**Comment by andremantaswow** [comment]
Can confirm this is skipping all my project files even if i have a lot of runtime warnings telling me to use named exports.

Is this supposed to work only for files that return JSX? It gives me warnings for non JSX things.

Eg `test.js`:
```
export default (
	a,
	b,
) => {
  return {
    prop1: a,
    prop2: b,
  }
};
```

```
$ npx @next/codemod name-default-component . --force
npx: installed 471 in 87.436s
WARNING: Git directory is not clean. Forcibly continuing.
Executing command: jscodeshift --verbose=2 --ignore-pattern=**/node_modules/** --ignore-pattern=**/.next/** --extensions=tsx,ts,jsx,js --parser=tsx --transform %USER%\AppData\Roaming\npm-cache\_npx\194052\node_modules\@next\codemod\transforms\name-default-component.js .
Processing 302 files...
Spawning 3 workers...
Sending 50 files to free worker...
Sending 50 files to free worker...
Sending 50 files to free worker...
...
All done.
Results:
0 errors
0 unmodified
302 skipped
0 ok
Time elapsed: 17.375seconds
